### DISCUSSION TITLE
Why is notional not following PEP8 naming conventions?

### DISCUSSION THREAD

**Discussion by FlorianWilhelm** [jheddings/notional#174] [General]
Is there a specific reason that the typical Python [naming convention PEP8](https://peps.python.org/pep-0008/#naming-conventions) is not used in notional? 

For instance the property `Markdown` in `notional.blocks` is uppercase and not `markdown`. Also in a `ToDo` block there is `IsChecked` which is rather Java CamlCase style and not `is_checked`. Other examples are `Title` vs. `title` and so on. 

That's just a question, maybe there is a reason for this :-)

**Comment by jheddings** [comment]
It's a completely fair question and there is a reason...

Notional uses `pydantic` to do data validation and serialization under the hood.  To help distinguish between "fields that are members of the API" and "fields that are provided by Notional," I used a different naming scheme for some properties.  The intent was to make it obvious and avoid possible collision in the event that the API added a field with the same name.

(I didn't say it was a particularly _good_ reason, but that was my initial thinking 😉  I'm certainly open to alternatives)

**Reply by FlorianWilhelm** [reply]
I see, thanks for the explanation, makes sense. I guess the alternative would habe been to have another API abstraction layer on top of the pydantic code, which increases the complexity maybe too much.
